generate.py

```diff
@@ -84,6 +84,7 @@ def generate(
         if idx_next == eos_id:
             return idx[:input_pos]  # include the EOS token
 
+    model.reset()
```

Comment:
I don't think the reset function should be in the generate function. Maybe move this to `main()`

Reply:
I feels like it is the clean up action of the generate function. We need to clean up after generation, so the next generation start from clean slate. 

Putting outside of generate function risks forgetting to call the reset function, when the generate function is imported to be used somewhere. For example, in the generate_adapter.py script.

Anyway, I have moved it to the main function. Let me know what you think.

Reply:
maybe just add instructions to how_to/inference.md describing the new function

Reply:
I don't think we need to add this to the howtos, as this is an implementation detail of the code. Maybe just add a comment for why we reset

Reply:
But I agree that reset should be part of the `main` and not `generate` or the model

Reply:
The rational for resetting the cache outside of generate is it is not part of generating text, so it shouldn't be part of the generate function. When benchmarking the times to generate responses, those benchmarks shouldn't include the time to reset cache afterwards.